datafusion/physical-plan/src/aggregates/group_values/group_value_row.rs

```diff
@@ -60,11 +61,13 @@ pub trait ArrayRowEq: Send + Sync {
     fn take_n(&mut self, n: usize) -> ArrayRef;
 }
 
+/// An implementation of [`ArrayRowEq`] for primitive types.
 pub struct PrimitiveGroupValueBuilder<T: ArrowPrimitiveType> {
     group_values: Vec<T::Native>,
     nulls: Vec<bool>,
```

Comment:
should be it a BooleanArray? so this null checks will be faster and in 1 place?

Reply:
You are foreshadowing :) This is an excellent point. @jayzhan211  and I are working on exactly this topic. https://github.com/apache/datafusion/pull/12623